### DISCUSSION TITLE
FIL/IPFS Hardware Acceleration

### DISCUSSION THREAD

**Discussion by dvo-orlando** [filecoin-project/rust-fil-proofs#1412] [Q&A]
Hi all, I'm working on an academic project to implement an algorithm in hardware on an IC. I'm excited about distributed computing and FIL/IPFS specifically. Are there any algorithms within this field that would benefit from hardware-acceleration?
I'm thinking along the lines of accelerated CID hashing or snark-compression; small algorithms or processes that are executed many many times on the scale of a massive distributed network.

**Comment by cryptonemo** [comment]
Pinging @dignifiedquire

**Comment by dignifiedquire** [comment]
Algorithms that might be of interested to have accelerated from a Filecoin perspective

- Poseidon Hash function
- Sha256 Hash function (already very optimized with AMD cpus, but might be interesting for non AMD CPUs)
- Elliptic curve operations as required by bellperson (zkSNARK proving system)
- FFT & MultiExp as used by bellperson (currently accelerated on the GPU, might be possible to do faster)
